cpp/src/neighbors/ivf_flat_build_float.cpp

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
In raft, these files are all autogenerated from a python script - which means that we don't need to manually keep these up to date https://github.com/rapidsai/raft/blob/branch-24.04/cpp/src/neighbors/ivf_flat_00_generate.py  Can we add a similar mechanism here so that we can automate updates to these files as the API changes?

I'm also wondering if we should move these explicit instantiations to a subdirectory to keep separate from the non-generated source code.

Reply:
I think it's a great idea to auto-generate the cpp source files but I don't think we should auto-generate the header files because those are what ultimately get exposed to the user and we want to make sure they are well documented and available in the source itself.